### PR TITLE
versions semigroup

### PR DESCRIPTION
**Pull request by coot**
I think the `Semigroup` instance is enough, `mempty` shouldn't be really used for `Versions`.

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Version.hs [hunk 1]
```diff
@@ -35,9 +36,22 @@ import Codec.SerialiseTerm
...
+-- >       :| [ (simpleSingletonVersions ...)
+-- >          , (simpleSingletonVersions ...)
+-- >          , ...
+-- >          ]
+-- >
+--
 newtype Versions vNum extra r = Versions
   { getVersions :: Map vNum (Sigma (Version extra r))
   }
+  deriving Semigroup via (Map vNum (Sigma (Version extra r)))
```
Comment (mrBliss) [inline ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Version.hs]: Wouldn't `deriving Semigroup` also have worked? Or `deriving newtype Semigroup`?

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Protocol/Handshake/Version.hs]: I am not sure.  But this is very explicit, so I like it over `deriving Semigroup`, `deriving newtype` - is equally good to me.

**Comment by manveru** [comment]
bors retry

**Pull request merged**